### ISSUE TITLE
Steamshovel no longer working with simple python server (Trac #2110)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2102]
Here at UMD, we've had a TV running steamshovel, taking data from a cache directory
and serving it out to any connected steamshovel.  We currently still have a version of steamshovel from
serveral years ago.  An attempt to use with trunk/steamshovel fails to get any events into the 
steamshovel for display.

Server is a simple python tornado server:
http://code.icecube.wisc.edu/svn/sandbox/blaufuss/i3tv_server/umd
Included instructions are there for running and connecting via steamshovel

I have a few example input files at UMD:
~blaufuss/cache/

Connection is reported, but no frames are available for display, or to play.

It seems some of the stubs of the socket: interface are there, but some pieces are missing.  Seems
to trace back to Hans's redo of shovelio

Our current working system has a very old source:
Working Copy Root Path: /opt/i3display/ofsw/src/steamshovel
URL: http://code.icecube.wisc.edu/svn/projects/steamshovel/trunk
Relative URL: ^/projects/steamshovel/trunk
Repository Root: http://code.icecube.wisc.edu/svn
Repository UUID: 16731396-06f5-0310-8873-f7f720988828
Revision: 129911
Node Kind: directory
Schedule: normal
Last Changed Author: david.schultz
Last Changed Rev: 129822
Last Changed Date: 2015-03-05 00:14:01 -0500 (Thu, 05 Mar 2015)

Migrated from https://code.icecube.wisc.edu/ticket/2110
```json
{
    "status": "closed", 
    "changetime": "2019-03-28T13:52:47", 
    "description": "Here at UMD, we've had a TV running steamshovel, taking data from a cache directory\nand serving it out to any connected steamshovel.  We currently still have a version of steamshovel from\nserveral years ago.  An attempt to use with trunk/steamshovel fails to get any events into the \nsteamshovel for display.\n\nServer is a simple python tornado server:\nhttp://code.icecube.wisc.edu/svn/sandbox/blaufuss/i3tv_server/umd\nIncluded instructions are there for running and connecting via steamshovel\n\nI have a few example input files at UMD:\n~blaufuss/cache/\n\nConnection is reported, but no frames are available for display, or to play.\n\nIt seems some of the stubs of the socket: interface are there, but some pieces are missing.  Seems\nto trace back to Hans's redo of shovelio\n\nOur current working system has a very old source:\nWorking Copy Root Path: /opt/i3display/ofsw/src/steamshovel\nURL: http://code.icecube.wisc.edu/svn/projects/steamshovel/trunk\nRelative URL: ^/projects/steamshovel/trunk\nRepository Root: http://code.icecube.wisc.edu/svn\nRepository UUID: 16731396-06f5-0310-8873-f7f720988828\nRevision: 129911\nNode Kind: directory\nSchedule: normal\nLast Changed Author: david.schultz\nLast Changed Rev: 129822\nLast Changed Date: 2015-03-05 00:14:01 -0500 (Thu, 05 Mar 2015)\n\n", 
    "reporter": "blaufuss", 
    "cc": "nega, david.schultz, blaufuss", 
    "resolution": "wontfix", 
    "_ts": "1553781167065708", 
    "component": "combo core", 
    "summary": "Steamshovel no longer working with simple python server", 
    "priority": "normal", 
    "keywords": "steamshovel sockets", 
    "time": "2017-11-03T20:04:01", 
    "milestone": "Summer Solstice 2019", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20171103T20:28:26: 
* david.schultz changed cc from "" to "cweaver"
* david.schultz commented:

> Probably the revision dropped socket support, or at least assumed a local file.
> 
> I'll guess this will be fixed whenever steamshovel moves to the new I3FrameSequence in dataio.  See #2074.
> 
> Note for Chris: dataio-shovel also chokes on a socket, though it just sits at the loading animation.  Seems to be waiting on a lock.

**Comment by icecube-trac** [comment]
Trac update at 20171105T17:18:57: blaufuss commented:

> What is the time scale for I3FrameSequence?  We have a request from Ralf to
> get something working at pole for new event displays (till now still running glshovel
> and i3tv, very old stuff).  We can work with local files for a short time
> but want a server-client setup working soon.
> 
> 
> 
>

**Comment by icecube-trac** [comment]
Trac update at 20171105T17:30:03: david.schultz commented:

> The time scale is unknown.  It turns out that it's a lot of work, and I don't have time for it.  So it's waiting on a new owner.

**Comment by icecube-trac** [comment]
Trac update at 20180922T09:00:14: 
* olivas changed milestone from "" to "Winter Solstice 2018"
* olivas changed cc from "cweaver" to "nega, david.schultz, blaufuss"
* olivas changed owner from "nega" to "olivas"
* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20180922T13:40:07: 
* nega commented:

> this has been long ago worked around. i'll confirm if its still an issue, and if not mark it as 'wontfix'. regardless blaufuss and i can work on it at our leisure 


* nega changed owner from "olivas" to "nega"

**Comment by icecube-trac** [comment]
Trac update at 20180923T02:16:28: olivas commented:

> Sounds good.  If it turns out to still be an issue and there's no guarantee it'll be fixed by the next milestone, can you also set the milestone to LTF?

**Comment by icecube-trac** [comment]
Trac update at 20180923T02:17:28: olivas commented:

> Comment from Slack "blaufuss [16:35] what is the work around?  other than just reading from a file?"

**Comment by icecube-trac** [comment]
Trac update at 20190327T14:53:35: 
* olivas changed milestone from "Vernal Equinox 2019" to "Summer Solstice 2019"
* olivas commented:

> Not sure if this is still an issue or not.  Pushing down the road.

**Comment by icecube-trac** [comment]
Trac update at 20190328T13:52:47: 
* nega changed resolution from "" to "wontfix"
* nega commented:

> closing as "wontfix" pending further need.


* nega changed status from "assigned" to "closed"
